spec/models/replication_transfer_spec.rb

```diff
@@ -35,6 +35,36 @@
       expect(r.update(stored: true)).to be true
       expect(r.bag.replicating_nodes).to include r.to_node
     end
+
+    context "when fixity_value set" do
+      let(:correct_fixity) { "98734723942304820348203840238402983409283408234" }
+      let(:fixity_alg) { Fabricate(:fixity_alg) }
+      let(:bag) {
+        bag = Fabricate(:bag_without_digests)
+        bag.message_digests << Fabricate.build(:message_digest, fixity_alg: fixity_alg, value: correct_fixity)
+        bag.save!
+        bag
+      }
+      let(:transfer) { Fabricate(:replication_transfer,
+        fixity_alg: fixity_alg,
+        bag: bag,
+        from_node: Fabricate(:local_node))
+      }
+      it "sets store_requested->true when correct" do
+        transfer.update!(fixity_value: correct_fixity)
+        expect(transfer.reload.store_requested).to be true
+      end
+      it "cancels with fixity_reject when incorrect" do
+        transfer.update!(fixity_value: "123908102831028301820398120398102830129830")
+        expect(transfer.reload.cancelled).to be true
+        expect(transfer.reload.cancel_reason).to eql("fixity_reject")
+      end
+      it "does not set store_requested when incorrect" do
+        transfer.update!(fixity_value: "102380183019283012830128309128301983")
+        expect(transfer.reload.store_requested).to be false
+      end
+    end
+
   end
```

Comment:
The short answer is probably not.  
The longer answer is I agree that this isn't clear and isn't that easy to follow.  This and the other hooks should probably be redesigned into a poro.  The controller should send the update hash to that poro, and then the poro will perform the correct kind of update.  [this comment copied below]